package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/405-wireguard

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/lua
+local site = require 'gluon.site'
+local util = require 'gluon.util'
+local uci = require('simple-uci').cursor()
+
+uci:delete('wireguard', 'mesh_vpn', 'user')
+
+local add_groups
+
+local function generate_wg_iface_name(peer)
+  return "mesh_wg_" .. peer
+end
+
+local function add_peer(group, name, config)
+  uci:section('network', 'interface', generate_wg_iface_name(name) .. 'm', {
+    type = 'wireguard',
+    ifname = generate_wg_iface_name(name),
+    proto = 'gluon_mesh',
+  })
+
+  uci:section('network', 'interface', generate_wg_iface_name(name) .. 's', {
+    ip6addr = 'fe80::2/64',
```

Comment:
This will prevent direct links between two such instances.

Reply:
we could generate the link-local address using the nodeid / main mac address of the device. That would mean that the multiple ireguard-interfaces on the same node would still have the same ll-address. This would resolve (2) from genofire and also allow for direct links between nodes.

Reply:
fixed the ip creation.